src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -69,16 +94,27 @@ export class FileDataSource implements IDataSource {
 	}
 
 	public hasChildren(tree: ITree, stat: FileStat | Model): boolean {
-		return stat instanceof Model || (stat instanceof FileStat && stat.isDirectory);
+		return stat instanceof Model || (stat instanceof FileStat && (stat.isDirectory || stat.isVirtualDirectory));
 	}
 
 	public getChildren(tree: ITree, stat: FileStat | Model): TPromise<FileStat[]> {
 		if (stat instanceof Model) {
 			return TPromise.as(stat.roots);
 		}
 
+		if (stat.isVirtualDirectory) {
+			this.resolveVirtualDirectories(stat.parent.children);
+			return TPromise.as(this.enableVirtualDirectories ? stat.children : []);
+		}
+
 		// Return early if stat is already resolved
 		if (stat.isDirectoryResolved) {
+			this.resolveVirtualDirectories(stat.children);
```

Comment:
Do you have to do it here? If yes - then why?

It looks weird when you do: "If it is already resolved then resolve it"